src/field/field255.rs

```diff
@@ -0,0 +1,483 @@
+// Copyright (c) 2023 ISRG
+// SPDX-License-Identifier: MPL-2.0
+
+//! Finite field arithmetic for `GF(2^255 - 19)`.
+
+use crate::{
+    codec::{CodecError, Decode, Encode},
+    field::{FieldElement, FieldElementVisitor, FieldError},
+};
+use fiat_crypto::curve25519_64::{
+    fiat_25519_add, fiat_25519_carry, fiat_25519_carry_mul, fiat_25519_from_bytes, fiat_25519_opp,
+    fiat_25519_sub, fiat_25519_tight_field_element, fiat_25519_to_bytes,
+};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::{
+    convert::TryFrom,
+    fmt::{self, Debug, Display, Formatter},
+    io::{Cursor, Read},
+    marker::PhantomData,
+    ops::{Add, AddAssign, Div, DivAssign, Mul, MulAssign, Neg, Sub, SubAssign},
+};
+use subtle::{Choice, ConstantTimeEq, ConstantTimeGreater, ConstantTimeLess};
+
+#[cfg(test)]
+pub(crate) mod integer;
+
+#[cfg(test)]
+use integer::Field255Integer;
+
+// `python3 -c "print(', '.join(hex(x) for x in (2**255-19).to_bytes(32, 'big')))"`
+const MODULUS_BIG_ENDIAN: [u8; 32] = [
+    0x7f, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff,
+    0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0xed,
+];
+
+/// `GF(2^255 - 19)`, a 255-bit field.
+#[derive(Clone, Copy, Default)]
```

Comment:
Is `Default` appropriate to derive? `fiat_25519_tight_field_element` is `[u64; 5]`, so I think the default value will be `[0u64; 5]`. Is that a valid `fiat_25519_tight_field_element` value?

Reply:
Yes, `Field255::zero()` produces the same thing. Though maybe I'll manually implement `Default` and dispatch to that, so it's clearer.